src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileFormat.cs

```diff
@@ -1027,5 +1028,84 @@ private static List<CentralTransitiveDependencyGroup> ReadProjectFileTransitiveD
             }
             return results;
         }
+
+        internal static IList<IAssetsLogMessage> ReadLogMessageArray(JsonElement json, string projectPath)
```

Comment:
the changes in this file appear unrelated to `CacheFileFormat`, from the PR's title.

Reply:
This was used by CacheFileFormat to read logmessages. I have now moved it into the custom converter class I have added to deserialize LogMessages.